firmware/library/L3_HAL/sddriver.hpp

```diff
@@ -0,0 +1,239 @@
+#pragma once
+
+#include <cstdint>
+#include "L1_Drivers/gpio.hpp"
+#include "L1_Drivers/ssp.hpp"
+#include "L2_Utilities/debug_print.hpp"
+
+// Relevant Datasheet: https://www.sdcard.org/downloads/pls/pdf/index.php?p=Part1_Physical_Layer_Simplified_Specification_Ver6.00.jpg&f=Part1_Physical_Layer_Simplified_Specification_Ver6.00.pdf&e=EN_SS1
+class SdInterface
+{
+ public:
+    // // @constructor     SdInterface()
+    // constexpr SdInterface() = 0;
+
+    // @destructor      ~SdInterface
+    virtual ~SdInterface() {};
+
+    // @struct          SdResponse_t
+    // @description     (see below)
+    // TODO(#151): Define a response frame struct since there are various
+    // responses that will be sent, depending on the command sent previously
+    struct SdResponse_t
```

Comment:
Remove the word, `Sd` from all types specified within the scope of the SD card interface. This is to improve readability and remove redudancy for the reader. For example what you have will look like this:

    SdInterface::SdResponse_t

Where as it would be cleaner to simply say:

    SdInterface::Response_t

Reading the first one, it reads as, "SD Interface SD response type" (SD said twice), where as the other one reads like "SD interface response type" (no more redundancy).

Reply:
Also, does this structure need to be packed? Is this an overlay for some bytes you are receiving from the SD card? If so, use SJ2_PACKED(struct).